product_tag_pricelist/models/product_pricelist_item.py

```diff
@@ -9,82 +9,88 @@ class PricelistItem(models.Model):
     _inherit = "product.pricelist.item"
 
     applied_on = fields.Selection(
-        selection_add=[("2a_product_tags", "Product Tags")],
+        selection_add=[
+            ("2a_product_tags", "Product Tags"),
+            ("2_product_category",),
+            ("1_product",),
+            ("0_product_variant",),
+        ],
         ondelete={"2a_product_tags": "set default"},
     )
     product_tag_ids = fields.Many2many(
-        comodel_name="product.template.tag", string="Product Tags"
-    )
-    # This field is stored to keep the sorting rules as two
-    # different queries are used
-    categ_name = fields.Char(
-        string="Categ Name", related="categ_id.complete_name", store=True
+        comodel_name="product.tag", string="Product Tags"
     )
 
-    @api.constrains("product_id", "product_tmpl_id", "categ_id", "product_tag_ids")
+    @api.constrains("applied_on", "product_tag_ids")
     def _check_product_consistency(self):
         for item in self:
             if item.applied_on == "2a_product_tags" and not item.product_tag_ids:
```

Comment:
why no filtered? please add on test this validation.
The message does not indicate which line is failing, this can be confusing when saving when multiple lines are being created, since it will not be known which one is giving an error, correct?